cli/src/test/scala-2.12/coursier/cli/CliFetchIntegrationTest.scala

```diff
@@ -380,4 +381,232 @@ class CliFetchIntegrationTest extends FlatSpec with CliTestLib {
       assert(!node.dependencies.exists(_.coord.startsWith("org.scala-lang:scala-library:2.11.")))
   }
 
+  /**
+    * Result:
+    * |└─ org.apache.commons:commons-compress:1.5
+    */
+  "local dep url" should "have coursier-fetch-test.jar" in withFile() {
+    (excludeFile, _) =>
+      withFile() {
+        (jsonFile, _) => {
+          // generate temp jar with fake content
+          val filename = "coursier-fetch-test.jar"
+          val jarFileContent = "tada"
+          val file = new File(filename)
+          val bw = new BufferedWriter(new FileWriter(file))
```

Comment:
You should be able to levearge `withFile` above, like `withFile("tada", filename, ext)`. Slight modification to `withFile` is required, so you won't need to do all the file operations here.